service/history/replication/task_processor.go

```diff
@@ -427,6 +428,32 @@ func (p *taskProcessorImpl) convertTaskToDLQTask(
 			},
 		}, nil
 
+	case enumsspb.REPLICATION_TASK_TYPE_SYNC_WORKFLOW_STATE_TASK:
+		taskAttributes := replicationTask.GetSyncWorkflowStateTaskAttributes()
+		executionInfo := taskAttributes.GetWorkflowState().GetExecutionInfo()
+		executionState := taskAttributes.GetWorkflowState().GetExecutionState()
+		currentVersionHistory, err := versionhistory.GetCurrentVersionHistory(executionInfo.GetVersionHistories())
+		if err != nil {
+			return nil, err
+		}
+		lastItem, err := versionhistory.GetLastVersionHistoryItem(currentVersionHistory)
+		if err != nil {
+			return nil, err
+		}
+
+		return &persistence.PutReplicationTaskToDLQRequest{
+			ShardID:           p.shard.GetShardID(),
+			SourceClusterName: p.sourceCluster,
+			TaskInfo: &persistencespb.ReplicationTaskInfo{
```

Comment:
I kinda wonder if we should generate a `tasks.SyncWorkflowStateTask` here and let persistence layer do the conversion to proto representation. Similar to how normal replication tasks are generated & persisted. 

If we were to have DLQ for other queues in the future, that should help unify the code path.